src/arrayMethodJoin.js

```diff
@@ -5,7 +5,30 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let joinedStr = '';
+    let separatorSymb = separator;
+
+    if (separator === undefined) {
+      separatorSymb = ',';
+    }
+
+    if (separator === null) {
+      separatorSymb = 'null';
+    }
```

Comment:
it'll be better to set default value to arguments of the function insteat of these checks